scripts/sensor/03b-time_frequency_csp.py

```diff
@@ -0,0 +1,866 @@
+    cbar = plt.colorbar(ax=ax, shrink=0.75, orientation='horizontal',
+                        mappable=img, )
+    cbar.set_ticks(lims)
+    cbar.set_ticklabels([round(lim, 1) for lim in lims])
+    cbar.ax.get_xaxis().set_label_coords(0.5, -0.3)
+    if value_type == "p":
+        cbar.set_label(r'$-\log_{10}(p)$')
+    if value_type == "t":
+        cbar.set_label(r't-value')
+
+
+def plot_t_and_p_values(
+    t_values: np.ndarray,
+    p_values: np.ndarray,
+    title: str,
+    subjects: List[str],
+    cfg,
+    tf: Tf
+) -> Figure:  # TODO : delete this function ?
+    """Plot t-values and either (p-values or clusters).
+
+    Parameters:
+    -----------
+    t-values
+    p_values
+    title
+    cfg
+    tf
+
+    Returns
+    -------
+    A figure with two subplot: t-values and p-values.
+    """
+    plt.close()
+    fig = plt.figure(figsize=(10, 5))
+    axes = [fig.add_subplot(121), fig.add_subplot(122)]
+
+    plot_axis_time_frequency_statistics(
+        ax=axes[0], array=t_values, cfg=cfg,
+        subjects=subjects, value_type="t", tf=tf)
+    plot_axis_time_frequency_statistics(
+        ax=axes[1], array=p_values, cfg=cfg,
+        subjects=subjects,  value_type="p", tf=tf)
+    plt.tight_layout()
+    fig.suptitle(title)
+    return fig
+
+
+@failsafe_run(on_error=on_error)
+def group_analysis(
+    subjects: List[str],
+    cfg,
+    pth: Pth,
+    tf: Tf,
+    report: Report,
+) -> None:
+    """Group analysis.
+
+    1. Average roc-auc scores:
+        - frequency 1D histogram
+        - time-frequency 2D color-map
+    2. Perform statistical tests
+        - plot t-values and p-values
+        - performs classic cluster permutation test
+
+    Returns
+    -------
+    None. Plots are saved in memory.
+    """
+    if len(subjects) == 0:
+        msg = "We cannot run a group analysis with just one subject."
+        logger.critical(gen_log_message(msg, step=3))
+
+    msg = "Running group analysis..."
+    logger.info(gen_log_message(msg, step=3))
+
+    ######################################################################
+    # 1. Average roc-auc scores across subjects
+
+    # just to obtain sfreq
+    epochs = read_epochs(pth.file(subject=subjects[0],
+                                  session=config.get_sessions(),
+                                  cfg=cfg))
+    sfreq = epochs.info['sfreq']
+
+    # Average Frequency analysis
+    all_freq_scores = load_and_average(pth.freq_scores, subjects=subjects)
+
+    # Calculating the 95% confidence intervals
+    # TODO: The confidence intervals indicated on the individual
+    # plot are not quite valid because
+    # they use only the std of the cross-validation scores.
+    # We should also divide by the square root.
+    all_freq_scores_std = load_and_average(
+        pth.freq_scores, subjects=subjects, average=False)
+    std_ = np.nanstd(all_freq_scores_std, axis=0, ddof=1)
+    conf_int = 1.96*std_ / np.sqrt(all_freq_scores_std.shape[0])
```

Comment:
where did you get this from?

Reply:
You should do the same thing that we do to average time decoding scores.

Reply:
I don't understand from which part of the code I should take inspiration from?

Reply:
here https://github.com/mne-tools/mne-bids-pipeline/blob/main/scripts/sensor/04-group_average.py#L158

Reply:
integrated with "bootstrap confidence intervals"